app/utils/ipfs.js

```diff
@@ -40,24 +56,44 @@ export async function saveText(text) {
   }
 
   const buf = Buffer.from(parsedText, 'utf8');
-  const saveResult = await getIpfsApi().add(buf);
 
-  await saveTextTheGraph(buf);
-  return saveResult.cid.toString();
+  const textIpfsS3 = {
+    content: buf,
+    encoding: 'utf8',
+  };
+
+  const [resultIpfsS3] = await Promise.all([
```

Comment:
Move this into a separate function where pass 2 arguments, buffer, and encoding.

This way it will be hard to miss changes if need to update that logic. From your own experience, you updated `saveFile` but missed that `saveText` also needs to be updated.